src/textual/debounce.py

```diff
@@ -0,0 +1,24 @@
+import asyncio
+from functools import wraps
+from typing import Callable, Optional
+
+from .app import App
+from .timer import Timer as TimerClass
+
+
+def debounce(timeout: float) -> Callable:
+    def decorator(f: Callable) -> Callable:
+        timer: Optional[TimerClass] = None
+
+        @wraps(f)
+        async def wrapper(self: App, *args, **kwargs) -> None:
+            nonlocal timer
+            if timer:
+                timer.stop_no_wait()
```

Comment:
Rather than stopping the timer, we could pause and resume it. This would be more efficient, since we wouldn't need to create tasks every time.

Reply:
I think we want to `reset()` it instead, as otherwise I think we'd need something like a timer to restart the original timer or some such.

I've also just done some testing and it looks like if we do the reset method, then it's the original `args` and `kwargs` that get passed to the handler when the timer fires. e.g. for the dictionary example, it'll be the first letter typed rather than the word.

Reply:
@willmcgugan not sure if you still have a view on this. A basic `reset` doesn't work. We'd need to replace the closure callback in the `Timer` but then we're not using a public api.